src/components/ReleaseMonitor/ApplicationFilter.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from 'react';
+import {
+  Menu,
+  MenuContent,
+  MenuList,
+  MenuItem,
+  MenuToggle,
+  Popper,
+  Badge,
+} from '@patternfly/react-core';
+import { FilterIcon } from '@patternfly/react-icons/dist/esm/icons/filter-icon';
+
+interface ApplicationFilterProps {
+  applications: string[];
+  onSelectionsChange: (selections: string[]) => void;
+}
+
+export const ApplicationFilter: React.FC<ApplicationFilterProps> = ({
```

Comment:
These components seems to be generic, Please a create a generic component for this and reuse it.

Reply:
About the Namespace and Applications, when you see the records, you will see the selections, currently the local dev env doesn't have records, so you can't see it.

You need to simulate some records with an existing components. The test case already cover this I believe.
See screenshot.
![Screenshot 2025-06-26 at 10 11 40](https://github.com/user-attachments/assets/82bdf0ae-a95c-46ec-8f39-60bbba71cf4c)
![Screenshot 2025-06-26 at 10 11 44](https://github.com/user-attachments/assets/db46482f-36cf-4163-af96-c8e92a4d42b0)
![Screenshot 2025-06-26 at 10 11 53](https://github.com/user-attachments/assets/38faf16b-6041-4df7-a9dd-248070925656)

Reply:
For filters, I put them on the ../Filter/generic for reuse.

Reply:
> About the Namespace and Applications, when you see the records, you will see the selections, currently the local dev env doesn't have records, so you can't see it.

What do you mean? I have releases in my namespace, so I should be able to see them. Right?

Reply:
Yeah, if there is no releases, then you won't see the Namespace names in filter.
E.G.
I can see namespaces A, B, C on the "Namespaces" page, but none of them has the releases yet. Then on the selection you can't see the A,B,C list there. But if there is any release on A, then you can see the Namespace list A.
That's why on this screenshot, it has the wlin-tenant list there, as I can see more than one namespace, but there is only one tenant wlin-tenant has the release(I manual triggered one).